src/GWallet.Backend/JsonRpcSharp.fs

```diff
@@ -73,23 +73,39 @@ module JsonRpcSharp =
             if not result.IsCompleted then
                 return! ReadPipeInternal reader stringBuilder
             else
+                reader.Complete()
                 return stringBuilder.ToString()
         }
 
-        let ReadPipe pipeReader =
-            ReadPipeInternal pipeReader (StringBuilder())
+        let ReadPipe pipeReader = async {
+            let! s = Async.Catch (ReadPipeInternal pipeReader (StringBuilder()))
+            return s
+        }
 
         let FillPipeAsync (socket: Socket) (writer: PipeWriter) = async {
-            // If incomplete messages become an issue here, consider reinstating the while/loop
-            // logic from the original C# source.  For now, assuming that every response is complete
-            // is working better than trying to handle potential incomplete responses.
-            let memory: Memory<byte> = writer.GetMemory minimumBufferSize
-            let! bytesReceived = ReceiveAsync socket memory SocketFlags.None
-            if bytesReceived > 0 then
-                writer.Advance bytesReceived
-            else
-                raise NoResponseReceivedAfterRequestException
+            let rec FillPipeAsyncHelper() = async {
+                let! bytesReceived = ReceiveAsync socket (writer.GetMemory minimumBufferSize) SocketFlags.None 
+                if bytesReceived > 0 then
+                    writer.Advance bytesReceived
+                    let! result = (writer.FlushAsync().AsTask() |> Async.AwaitTask)
+                    // here we should to check if socket is open and data is avaliable there
+                    // but socket.Available can throw exceptions, so use try-catch block
+                    let socketStatus = 
+                        try 
+                            socket.Available
+                        with
```

Comment:
can you please specify the type of the exception that you're expecting here? I'm guessing it's only one type (not just `System.Exception` but a more specific one); thanks!

Reply:
2 types are expected:

```SocketException``` - An error occurred when attempting to access the socket.
```ObjectDisposedException``` - The Socket has been closed.
I'm not sure how to catch them. The first one is just an error of course, but the second one? Can a socket become closed automatically? Or it is impossible...

Reply:
is it a `System.ObjectDisposedException: Cannot access a disposed object. Object name: 'System.Net.Sockets.NetworkStream'.`? If yes, that's another instance of this other bug I think: https://gitlab.com/DiginexGlobal/geewallet/issues/25

Can you paste full stacktraces of these 2 exceptions here?